src/vcCamera.cpp

```diff
@@ -56,7 +56,7 @@ void vcCamera_UpdateSmoothing(vcCamera *pCamera, vcCameraInput *pCamInput, doubl
       pCamInput->smoothRotation -= step;
 
       pCamera->headingPitch.x = udMod(pCamera->headingPitch.x, UD_2PI);
-      pCamera->headingPitch.y = udClamp(pCamera->headingPitch.y, -UD_HALF_PI, UD_HALF_PI);
+      pCamera->headingPitch.y = udMod(pCamera->headingPitch.y, UD_2PI);
```

Comment:
Pitch should be UD_HALF_PI (straight up) to -UD_HALF_PI (straight down), otherwise the camera will flip over and be 'upside-down'

Reply:
I was wrong, @asunawu did a demo and this is correct.